### PR TITLE
Reduce doc build warnings, fix tutorials

### PR DESCRIPTION
**Pull request by charleskawczynski**
# Description

UPDATE: This PR is being split off into smaller ones since the doc build was taking too long. See #1219, #1218.

 - Fixes diagnostics API docs
 - Adds `dry_rayleigh_benard.md` to tutorials pages array, and fixes .jl comments

This also does some pretty large sweeping doc fixes.

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by simonbyrne** [comment]
Is `dry_raileigh_benard` really a tutorial, or should it be an experiment?

**Comment by charleskawczynski** [comment]
I think it’s a tutorial, and not an experiment, cc @akshaysridhar

File: tutorials/Atmos/heldsuarez.jl [hunk 1]
```diff
@@ -192,23 +189,23 @@ model = AtmosModel{FT}(
...
-n_vert = 2                            ## vertical element number
+poly_order = 5;                        ## discontinuous Galerkin polynomial order
+n_horz = 2;                            ## horizontal element number
+n_vert = 2;                            ## vertical element number
 resolution = (n_horz, n_vert)
-n_days = 1                            ## experiment day number
-timestart = FT(0)                     ## start time (s)
-timeend = FT(n_days * day(param_set)) ## end time (s)
-nothing # hide
+n_days = 0.1;                          ## experiment day number
```
Comment (charleskawczynski) [inline tutorials/Atmos/heldsuarez.jl:205]: @bischtob please confirm that this is okay for the tutorial

File: tutorials/Atmos/dry_rayleigh_benard.jl [hunk 2]
```diff
@@ -172,8 +179,9 @@ function config_problem(FT, N, resolution, xmax, ymax, zmax)
     return config
 end
 
+# Define diagnostics configuration kernel
 function config_diagnostics(driver_config)
-    interval = "10000steps"
+    interval = "100steps"
```
Comment (charleskawczynski) [inline tutorials/Atmos/dry_rayleigh_benard.jl:184]: @akshaysridhar please confirm that this is okay for the tutorial

File: tutorials/Atmos/risingbubble.jl [hunk 3]
```diff
@@ -320,15 +295,15 @@ function main()
     ymax = FT(500)
     zmax = FT(10000)
     t0 = FT(0)
-    timeend = FT(1000)
+    timeend = FT(100)
```
Comment (charleskawczynski) [inline tutorials/Atmos/risingbubble.jl:298]: @akshaysridhar please confirm that this is okay for the tutorial

**Comment by charleskawczynski** [comment]
The doc build is taking a long time. Going to split this PR up into smaller parts.

**Comment by charleskawczynski** [comment]
Superseded by #1244

**Pull request closed without merging**